### PR TITLE
extend i18n.get with macro replacement capabilities

### LINKED ISSUE #1255: App share message says {{app.name}} instead of actual app name
Just made an app. Tried to share it via text message. Message says `{{app.name}}` instead of actual app name.

screenshot_2015-02-20-12-20-57

### PR DESCRIPTION
**Pull request by Pomax**
fixes #1255 by adding macro capabilities to i18n.get

### CODE REVIEW

File: lib/i18n.js [hunk 1]
```diff
@@ -79,7 +79,15 @@ function Localize () {
...
             }
         }
-        return dict || defaultLang || key;
+        var value = dict || defaultLang || key;
+        if (!!value) {
+            var macroReplace = function (pattern, term) {
+                var replacement = !!macros ? macros[term] : false;
+                return replacement ? replacement : pattern;
+            };
+            value = value.replace(/{{\s*([^}\s]+)\s*}}/g, macroReplace);
```
Comment (secretrobotron) [inline lib/i18n.js]: You sure you want to leave out whitespace from the group? e.g. `app name` won't work, but `appname` or `app.name` will.

**Comment by secretrobotron** [comment]
Just spoke with @k88hudson a bit about this. The problematic message was altered to remove its macro (https://github.com/mozilla/webmaker-app/blob/master/locale/en_US/webmaker-app.json#L298), and the others don't use the `i18n` lib programmatically.

Her advice was to close this and avoid putting macros in those strings.

@Pomax make sense?

File: lib/i18n.js (hunk 1 above)
Comment (Pomax) [inline lib/i18n.js]: good point

**Comment by Pomax** [comment]
yeah that works for me!

**Pull request closed without merging**